django/core/cache/backends/db.py

```diff
@@ -202,15 +202,29 @@ def _base_set(self, mode, key, value, timeout=DEFAULT_TIMEOUT):
                 return True
 
     def delete(self, key, version=None):
-        key = self.make_key(key, version=version)
-        self.validate_key(key)
+        self.delete_many([key], version)
+
+    def delete_many(self, keys, version=None):
+        key_map = {self.make_key(key, version): key for key in keys}
 
         db = router.db_for_write(self.cache_model_class)
         connection = connections[db]
+        quote_name = connection.ops.quote_name
         table = connection.ops.quote_name(self._table)
 
         with connection.cursor() as cursor:
-            cursor.execute("DELETE FROM %s WHERE cache_key = %%s" % table, [key])
+            if len(key_map) == 1:
+                format_string = '= %s'
+            else:
+                format_string = 'IN(%s' + ', %s' * (len(key_map) - 1) + ')'
```

Comment:
This can be simplified and put in place of `format_string` below:
```python
'= %s' if len(key_map) == 1 else 'IN(%s)' % ', '.join(['%s'] * len(key_map))
```

Reply:
I don't see why we need to special case 1 item with `=` when `IN` still works with a tuple of 1 item. Also please add a space between `IN` and the opening bracket `(` - it's an operator not a function.

Reply:
(Interestingly @pope1ni  as I noted below, if this is left as a special case we need another test to get coverage - but if we compress to the one line ternary form you suggest, the whole line shows up as covered, cheating the metrics. This is why I don't like one line ternary expressions 😁 )

Reply:
Interesting... Hadn't thought of the coverage issue.
I tend to prefer them because it is less verbose and I don't like lots of unnecessary temporaries.

Regarding the special case for 1 item, I refer you to: https://github.com/django/django/pull/10246#issuecomment-408878911

Reply:
Ah right. I'd like a citation on that from @charettes though, I don't see how it doesn't "optimize" 🤔

Reply:
https://www.1keydata.com/sql/sqlin.html
```
The number of values in the parenthesis can be one or more, with each values separated by comma. Values can be numerical or string characters. If there is only one value inside the parenthesis, this commend is equivalent to,

WHERE "column_name" = 'value1'
```
I think There is no difference.